client/verta/verta/_dataset_versioning/dataset.py

```diff
@@ -57,15 +57,54 @@ def _generate_default_name(cls):
 
     @classmethod
     def _get_proto_by_id(cls, conn, id):
-        raise NotImplementedError
+        Message = _DatasetService.GetDatasetById
+        msg = Message(id=id)
+        endpoint = "/api/v1/modeldb/dataset/getDatasetById"
+        response = conn.make_proto_request("GET", endpoint, params=msg)
+        return conn.maybe_proto_response(response, Message.Response).dataset
 
     @classmethod
     def _get_proto_by_name(cls, conn, name, workspace):
-        raise NotImplementedError
+        Message = _DatasetService.GetDatasetByName
+        msg = Message(name=name, workspace_name=workspace)
+        endpoint = "/api/v1/modeldb/dataset/getDatasetByName"
+        proto_response = conn.make_proto_request("GET", endpoint, params=msg)
+        response = conn.maybe_proto_response(proto_response, Message.Response)
+
+        if workspace is None or response.HasField("dataset_by_user"):
+            return response.dataset_by_user
+        elif hasattr(response, "shared_datasets") and response.shared_datasets:
+            return response.shared_datasets[0]
+        else:
+            return None
 
     @classmethod
     def _create_proto_internal(cls, conn, ctx, name, desc=None, tags=None, attrs=None, date_created=None, public_within_org=None):
-        raise NotImplementedError
+        Message = _DatasetService.CreateDataset
+        msg = Message(name=name, description=desc, tags=tags, attributes=attrs, workspace_name=ctx.workspace_name)
+        endpoint = "/api/v1/modeldb/dataset/createDataset"
```

Comment:
Move this line a bit closer to the `make_proto_request()` call, please!

Reply:
You mean the endpoint line? The msg line must precede the "if public_within_org" block, because under the "if" block a field of the message is set.

Reply:
Yep, just the `endpoint` line. Thanks!